centreon/src/Core/Resources/Infrastructure/API/FindResourcesByParent/FindResourcesByParentController.php

```diff
@@ -0,0 +1,94 @@
+
+use Centreon\Application\Controller\AbstractController;
+use Centreon\Domain\Monitoring\ResourceFilter;
+use Core\Resources\Application\UseCase\FindResourcesByParent\FindResourcesByParent;
+use Core\Resources\Infrastructure\API\FindResources\FindResourcesRequestValidator as RequestValidator;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @phpstan-import-type _RequestParameters from RequestValidator
+ */
+final class FindResourcesByParentController extends AbstractController
+{
+    private const HOST_RESOURCE_TYPE = 'host';
+
+    public function __construct(private readonly RequestValidator $validator)
+    {
+    }
+
+    /**
+     * @param FindResourcesByParent $useCase
+     * @param FindResourcesByParentPresenter $presenter
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function __invoke(
+        FindResourcesByParent $useCase,
+        FindResourcesByParentPresenter $presenter,
+        Request $request
+    ): Response
+    {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+
+        $filter = $this->validator->validateAndRetrieveRequestParameters($request->query->all());
+
+        $useCase($presenter, $this->createResourceFilter($filter));
+
+        return $presenter->show();
+    }
+
+    /**
+     * @param _RequestParameters $filter
+     *
+     * @return ResourceFilter
+     */
+    private function createResourceFilter(array $filter): ResourceFilter
+    {
+        return (new ResourceFilter())
+            ->setStates($filter[RequestValidator::PARAM_STATES])
+            ->setStatuses($filter[RequestValidator::PARAM_STATUSES])
+            ->setStatusTypes($filter[RequestValidator::PARAM_STATUS_TYPES])
+            ->setServicegroupNames($filter[RequestValidator::PARAM_SERVICEGROUP_NAMES])
+            ->setServiceCategoryNames($filter[RequestValidator::PARAM_SERVICE_CATEGORY_NAMES])
+            ->setServiceSeverityNames($filter[RequestValidator::PARAM_SERVICE_SEVERITY_NAMES])
+            ->setServiceSeverityLevels($filter[RequestValidator::PARAM_SERVICE_SEVERITY_LEVELS])
+            ->setHostgroupNames($filter[RequestValidator::PARAM_HOSTGROUP_NAMES])
+            ->setHostCategoryNames($filter[RequestValidator::PARAM_HOST_CATEGORY_NAMES])
+            ->setHostSeverityNames($filter[RequestValidator::PARAM_HOST_SEVERITY_NAMES])
+            ->setMonitoringServerNames($filter[RequestValidator::PARAM_MONITORING_SERVER_NAMES])
+            ->setHostSeverityLevels($filter[RequestValidator::PARAM_HOST_SEVERITY_LEVELS])
+            ->setOnlyWithPerformanceData($filter[RequestValidator::PARAM_RESOURCES_ON_PERFORMANCE_DATA_AVAILABILITY])
+            ->setTypes(
+                array_filter(
+                    $this->validator->getResourceTypes(),
+                    static fn (string $resourceType) => $resourceType !== self::HOST_RESOURCE_TYPE
+                )
+            );
+    }
```

Comment:
Auto-documented code : readibility.

```suggestion
            ->setTypes(
                array_filter(
                    $this->validator->getResourceTypes(),
                    $this->removeHostFromTypes(...)
                )
            );
    }

    private function removeHostFromTypes(string $resourceType): bool
    {
        return $resourceType !== self::HOST_RESOURCE_TYPE;
    }
```